app/api/v2/models.py

```diff
@@ -0,0 +1,298 @@
+        db_obj=Dbase()
+        self.conn=db_obj.connection_to_Dbase()
+        cursor=self.conn.cursor()
+        cursor.execute("INSERT INTO parties(party_name,hqaddress,logoUrl)\
+            VALUES (%s, %s, %s)",
+            (self.party_name,self.hqaddress,self.logoUrl))
+            
+        self.conn.commit()
+        self.conn.close()
+
+    def get_all_parties(self):
+        db_obj = Dbase()
+        self.conn = db_obj.connection_to_Dbase()
+        cursor = self.conn.cursor()
+        cursor.execute("SELECT * FROM parties")
+        result = cursor.fetchall()
+        partylist = []
+
+        for Party in result:
+            single_party = {}
+            single_party['party_id'] = Party[0]
+            single_party["party_name"] = Party[1]
+            single_party["hqaddress"]=Party[2]
+            single_party['logoUrl'] = Party[3]
+            partylist.append(single_party)
+
+        self.conn.close()
+        return partylist 
+
+class Office(Dbase):
+    def __init__(self, data=None):
+        if data:
+            self.office_name= data['office_name']
+            self.office_type= data['office_type']
+
+    def save_office(self):
+        db_obj=Dbase()
+        self.conn=db_obj.connection_to_Dbase()
+        cursor=self.conn.cursor()
+        cursor.execute("INSERT INTO offices(office_name,office_type)\
+            VALUES (%s, %s)",
+            (self.office_name,self.office_type))
+            
+        self.conn.commit()
+        self.conn.close()        
+       
+      
+    def get_all_offices(self):
+        db_obj = Dbase()
+        self.conn = db_obj.connection_to_Dbase()
+        cursor = self.conn.cursor()
+        cursor.execute("SELECT * FROM offices")
+        result = cursor.fetchall()
+        officelist = []
+
+        for office in result:
+            single_office = {}
+            single_office['office_id'] = office[0]
+            single_office["office_name"] = office[1]
+            single_office["office_type"]=office[2]
+            officelist.append(single_office)
+
+        self.conn.close()
+        return officelist  
+
+class Candidate(Dbase):
+    def __init__(self, data=None):
+        if data:
+            self.user_id= data['user_id']
+            self.office_id=data['office_id']
+            db_obj=Dbase()
+            self.conn=db_obj.connection_to_Dbase()
+
+
+    def save(self):
+        db_obj=Dbase()
+        self.conn=db_obj.connection_to_Dbase()
+        cursor=self.conn.cursor()
+        cursor.execute("INSERT INTO candidates(office_id,user_id)\
+            VALUES (%s, %s)",
+            (self.office_id,self.user_id))
+            
+        self.conn.commit()
+        self.conn.close() 
+            
+    def get_all_candidates(self):
+        db_obj = Dbase()
+        self.conn = db_obj.connection_to_Dbase()
+        cursor = self.conn.cursor()
+        cursor.execute("SELECT candidates.candidate_id,users.user_id,users.username,offices.office_id,offices.office_name\
```

Comment:
The same for this as well. You have multiple INSERT statements. You can remove the repititions